modules/apps/forms-and-workflow/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/portlet/CalendarPortlet.java

```diff
@@ -257,11 +257,34 @@ public void invokeTransition(
 
 		int status = ParamUtil.getInteger(actionRequest, "status");
 
+		long startTime = ParamUtil.getLong(actionRequest, "startTime");
+
+		boolean updateInstance = ParamUtil.getBoolean(
+			actionRequest, "updateInstance");
+
+		boolean allFollowing = ParamUtil.getBoolean(
+			actionRequest, "allFollowing");
+
 		ServiceContext serviceContext = ServiceContextFactory.getInstance(
 			CalendarBooking.class.getName(), actionRequest);
 
-		_calendarBookingService.invokeTransition(
-			calendarBookingId, status, serviceContext);
+		CalendarBooking calendarBooking =
+			_calendarBookingService.invokeTransition(
+				calendarBookingId, status, startTime, updateInstance,
+				allFollowing, serviceContext);
+
+		String redirect = getRedirect(actionRequest, actionResponse);
+
+		if (calendarBooking.getCalendarBookingId() != calendarBookingId) {
+			redirect = StringUtil.replace(
+				redirect, "/-/calendar/event/" + calendarBookingId,
+				"/-/calendar/event/" + calendarBooking.getCalendarBookingId());
```

Comment:
I researched and found no better solution for editing the URL here :disappointed: Yet, manually editing URLs caused us some bugs in the past (cf. https://github.com/natocesarrego/liferay-portal/commit/fdb1458b23c216f87b77580f837ef5958e55c97c)

So, my suggestion is to create a new render URL instead of "editing" the redirect. I commited it here: https://github.com/brandizzi/liferay-portal/pull/323/commits/aa9d186e23b0d815c475838458b74c4759c846cd